### ISSUE TITLE
jinja[invalid] error on loop when using attribute from "item"

### ISSUE THREAD

**Issue by jsabater** [ansible/ansible-lint#3201]
##### Summary

I have a task in a playbook that loops through a list of items in a variable and uses a conditional. This conditional uses an attribute of each item of the loop. `ansible-lint` complains with the following error, but the playbook works just fine:

```
jinja[invalid]: Unhandled exception: a bytes-like object is required, not 'AnsibleUndefined'
```

I am not sure whether this is actually a bug of `ansible-lint` or not, as I have not been able to find any way to format the condition so that it does not produce the abovementioned error.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

Running Debian 11 Bullseye with the latest version of Ansible and `ansible-lint` in a virtual environment on Python 3.9 created using `venv`.

```console (paste below)
$ ansible-lint --version
ansible-lint 6.14.2 using ansible 2.14.3
```
This is the task:

```console (paste below)
- name: Loop through the list of URL with priority greater than or equal to 0.5 and load them
  ansible.builtin.uri:
    url: "{{ item.loc }}"
    method: GET
    status_code: 200
    timeout: 5
  loop: "{{ sitemap.content | sitemapxml2dict }}"
  when: item.priority | float > 0.4
  delegate_to: localhost
```

**Comment by ajinkyau** [comment]
@jsabater I am unable to reproduce this issue using the current ansible-lint version and also via the virtual environment setup that has been mentioned above.
